src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -506,7 +507,8 @@ private void InitNuspecOutputPath(IPackTaskRequest<IMSBuildItem> request, PackAr
                 var currentPath = targetPath;
                 var fileName = Path.GetFileName(sourcePath);
                 if (!Path.GetExtension(fileName)
-                    .Equals(Path.GetExtension(targetPath), StringComparison.OrdinalIgnoreCase))
+                    .Equals(Path.GetExtension(targetPath), StringComparison.OrdinalIgnoreCase)
+                    || Path.GetExtension(fileName) == String.Empty)
```

Comment:
change to string.IsNullOrEmpty(Path.GetExtension(fileName)) and make that to be the first condition check